frontend/javascripts/libs/trackball_controls.ts

```diff
@@ -211,9 +211,12 @@ const TrackballControls = function (
         angle *= _this.rotateSpeed;
         quaternion.setFromAxisAngle(axis, -angle);
 
-        _eye.applyQuaternion(quaternion);
+        _eye.sub(_this.flycamPos).applyQuaternion(quaternion).add(_this.flycamPos);
 
         _this.object.up.applyQuaternion(quaternion);
+        _this.object.position.sub(_this.flycamPos).applyQuaternion(quaternion).add(_this.flycamPos);
+
+        _this.target.sub(_this.flycamPos).applyQuaternion(quaternion).add(_this.flycamPos);
```

Comment:
The rotation only works if the flycam position is centered in the 3D viewport. If the user pans a bit and then rotates, the rotation goes wild. The code here might be responsible for this.